### DISCUSSION TITLE
MQTT switch ESPEASY STATO UNKNOWN

### DISCUSSION THREAD

**Discussion by jimborm** [indomus/forum#1941] [HUB: Home Assistant]
Ho un nodeMCU (ESP8266 con firmware ESP Easy) a cui ho collegato il GPIO12 ad un relè che chiude il contatto del tasto del citofono che a sua volta pilota un cancello scorrevole.
Su Home Assistant ho configurato lo switch MQTT  e un'automazione che mi forza ad off lo stato dopo 30 secondi **ma non riesco a vedere lo stato dello switch nonostante queste configurazioni** 

<img width="644" alt="Schermata 2022-09-24 alle 19 52 55" src="https://user-images.githubusercontent.com/52866001/192112813-dc087650-eda0-4a91-b7fc-ee2f5f09674d.png">
<img width="549" alt="Schermata 2022-09-24 alle 19 54 23" src="https://user-images.githubusercontent.com/52866001/192112828-374550ea-bb0c-45f1-8522-394159877b92.png">
<img width="884" alt="Schermata 2022-09-24 alle 19 55 09" src="https://user-images.githubusercontent.com/52866001/192112835-9eb7063b-f355-4f46-be66-b0ef935841ad.png">
<img width="1373" alt="Schermata 2022-09-24 alle 19 53 24" src="https://user-images.githubusercontent.com/52866001/192112886-d53d3f99-5457-476d-9c2f-0f5fdbdcfcd1.png">

**Comment by snakuzzo** [comment]
Probabile tu abbia sbagliato lo `state_topic`...oppure devi sistemare la gestione del message di quel topic con un template.
Inoltre, se il firmware te lo consente, utilizza il LWT topic e quindi l' `availability_topic` nella definizione dell'entità.

Verifica cosa passa sulla coda del broker MQTT. Puoi farlo con i mosquitto client o, se ti torna comodo, con MQTT Explorer.

Posta qui qualche esempio di topic e message che passano sulla coda (USA IL MARKDOWN!)

**Reply by jimborm** [reply]
Non so se questo firmwareESP-EASY che ho caricato sulla board ESP abbia la possibilita di inserire il topic sullo stato o sia un topic che di default viene popolato. Andando negli Advanced Setting dell'interfaccia grafica dell'ESP vedo questi parametri
![image](https://github.com/user-attachments/assets/542d3996-c6f8-4576-b2d5-ff43a860bcca)
Quando sulla dashboard attivo il tasto con MQTT Explorer vedo questo 
```
{
"log": "GPIO 15 Pulsed for 500 mS",
"plugin": 1,
"pin": 15,
"mode": "output",
"state": 0
}
```
![image](https://github.com/user-attachments/assets/c6b66968-547d-4f30-939e-0c24ebf53b83)

**Reply by snakuzzo** [reply]
Sulla dashboard di cosa ? Di Home Assistant ? Non c'entra proprio nulla. E' l'ultima cosa che devi verificare. Ti devi focalizzare sul funzionamento di esp easy, devi capire com'è strutturato a livello MQTT. Quali sono i topic e i payload da usare per impartire un comando e quali sono i topic di stato.

**Reply by jimborm** [reply]
Il primo screenshot che ti ho postato sopra è proprio della pagina di esp easy dove mi sembra ci siano i settaggi per MQTT, ma li non vedo i settaggi dell'availability_topic. Inoltre nella schermata principale di Es Easy ci sono i settaggi del broker ma niente altro sui topic.
![image](https://github.com/user-attachments/assets/736c9f07-0a26-49c7-9661-5db2b363dce0)
Non so sinceramente se di default ci sia un topic per indicare lo stato del device. Pensavo di poterlo recuperare dal MQTT Explorer ma anche qui non li vedo. 
Possibile  che non li manda? Il protocollo che usa è OpenHAB MQTT

**Reply by snakuzzo** [reply]
Forse non mi sono spiegato.
Nello screenshot dell'altro giorno ci sono le informazioni basiche dove imposti come dev'essere templatizzato (lasciami passare il termine) quello che passa sulla coda del broker mqtt.
Devi capire cosa passa sul broker quando fai qualcosa sul citofono (quando suona, quando apri il portone, etc) e devi capire come impartire i comandi sempre via MQTT (leggi la documentazione ufficiale -> https://espeasy.readthedocs.io/en/latest/Controller/C005.html#c005-page).
Fatto questo, capirai come definire le entità MQTT su HA.
L'ultimo screenshot che hai postato invece non serve a niente.

**Comment by jimborm** [comment]
Scusami ma io non riesco a capire. Ho visto il link che mi hai postato ma non riesco a trovare le info utili a capire se, ed eventualmente come, posso impostare il nodeMCU a popolare un topic che mi indichi lo stato del GPIO14 . 
Se nei settaggi del firmware per MQTT vedo solo quei due settaggi che ti ho messo nel primo screen (subscribe template e Publish template)  come potrei fare per far pubblicare un topic sullo stato ?
Tanto piu che sul MQTT Explorer per quel NodeMCU vedo solo questi topic
![image](https://github.com/user-attachments/assets/df2ee251-52cd-429d-895d-1e7b0b61508a)

**Reply by snakuzzo** [reply]
Non ne usciamo così, perché non stai afferrando quello che ti ho scritto. Io da qui non saprei aiutarti, perché non uso e conosco espeasy e non ho tempo di mettermi lì a capire cosa succede nel tuo caso (anche perché non è che ho il tuo device sotto mano). Se posso consigliarti una cosa, butta via espeasy (non so perché hai scelto sta roba) e usa esphome

**Comment by jimborm** [comment]
Ciao questo device è il primo che ho configurato tantissimo tempo fa su indicazione di un collega (ingegnere esperto smanettone) che praticamente mi ha seguito passo passo ma lo usavo con domoticz ai tempi. Cmq i dispositivo cel'ho messo in una posizione difficile da raggiungere quindi materialmente mi rimane complicato prenderlo per fare prove. Pensi che da wifi posso convertirlo/riprogrammarlo come dici tu?

**Reply by snakuzzo** [reply]
https://esphome.io/guides/migrate_espeasy.html

**Reply by jimborm** [reply]
OK quindi per ricapitolare con ESPHome posso crearmi la configurazione per il mio NodeMCU (8266) che abbia queste caratteristiche:
- Connessione al wifi
- 4 switch come quelli configurati attualmente su espeasy
![image](https://github.com/user-attachments/assets/40546d6a-ee67-4161-8ddb-1708d5754186)
- Configurazione MQTT per comunicare con il BROKER in modo che possa:
               - ricevere il comando di accensione (adesso invio questo "Pulse,13,0,200"). Questo con il topic payload
               -  inviare lo stato dei 4 switch. Questo con l'availability_topic
Ho riassunto tutto?

Un dubbio però adesso se devo cambiare qualcosa vado sulla pagina web del NodeMCU e cambio/configuro il device lo restarto se serve. Con ESP HOME per fare modifiche al dispositivo che procedura si deve usare? Devo modificare il file yaml di configurazione su home assistant e poi rimandarlo al device dall'interfaccia di esphome?
![image](https://github.com/user-attachments/assets/328ae21d-dca9-4643-bf36-d5f419590750)

**Reply by snakuzzo** [reply]
MQTT non è necessario. Se ti occorre puoi abilitarlo, altrimenti puoi anche non usarlo visto che HA e Esphome comunicano via API alla perfezione.
Se devi modificare qualcosa, fai una modifica al file YAML, fai la Validate e ricompili al volo con aggiornamento via OTA (con la dashboard di esphome poi è tutto molto semplice).
